EventFilter/HcalRawToDigi/plugins/PackerHelp.h

```diff
@@ -0,0 +1,480 @@
+
+    AMC13Header = 0;
+    AMC13Header |= (AMC13HeaderSpec::FIXED_LSB&AMC13HeaderSpec::MASK_FIXED_LSB)<<AMC13HeaderSpec::OFFSET_FIXED_LSB;
+    AMC13Header |= ((OrbitNum+1)&AMC13HeaderSpec::MASK_ORN)<<AMC13HeaderSpec::OFFSET_ORN;
+    AMC13Header |= (uint64_t(0x0)&AMC13HeaderSpec::MASK_RESERVED)<<AMC13HeaderSpec::OFFSET_RESERVED;
+    AMC13Header |= (uint64_t(0x0)&AMC13HeaderSpec::MASK_NAMC)<<AMC13HeaderSpec::OFFSET_NAMC;
+    AMC13Header |= (uint64_t(0x0)&AMC13HeaderSpec::MASK_RES)<<AMC13HeaderSpec::OFFSET_RES;
+    AMC13Header |= (uint64_t(0x1)&AMC13HeaderSpec::MASK_UFOV)<<AMC13HeaderSpec::OFFSET_UFOV;
+
+  }
+
+  void setNAMC(uint64_t NAMC){
+    AMC13Header |= (NAMC&AMC13HeaderSpec::MASK_NAMC)<<AMC13HeaderSpec::OFFSET_NAMC;
+  }
+
+  void addAMCHeader( uint64_t crate , uint64_t slot , uint64_t AMCsize , uint64_t presamples = 10 , uint64_t blockNum = 0 ){
+
+    uint64_t header = 0 ;
+    header |= (crate&AMCHeaderSpec::MASK_CRATE_ID)<<AMCHeaderSpec::OFFSET_CRATE_ID ;
+    header |= (slot&AMCHeaderSpec::MASK_SLOT_ID)<<AMCHeaderSpec::OFFSET_SLOT_ID ;
+    header |= (presamples&AMCHeaderSpec::MASK_PRESAMPLES)<<AMCHeaderSpec::OFFSET_PRESAMPLES ; // boardId
+    header |= (slot&AMCHeaderSpec::MASK_AMC_NO)<<AMCHeaderSpec::OFFSET_AMC_NO ; // AMC no.
+    header |= (blockNum&AMCHeaderSpec::MASK_BLK_NO)<<AMCHeaderSpec::OFFSET_BLK_NO ; // Block No.
+    header |= (AMCsize&AMCHeaderSpec::MASK_AMCN_SIZE)<<AMCHeaderSpec::OFFSET_AMCN_SIZE ; // size 
+    header |= uint64_t(0x1)<<AMCHeaderSpec::OFFSET_C ; // CRC is valid
+    header |= uint64_t(0x1)<<AMCHeaderSpec::OFFSET_V ; // EvN, BcN match
+    header |= uint64_t(0x1)<<AMCHeaderSpec::OFFSET_P ; // Present, header is only made if data is present
+    header |= uint64_t(0x1)<<AMCHeaderSpec::OFFSET_E ; // Enabled, header is only made if AMC is enabled
+    header |= uint64_t(0x0)<<AMCHeaderSpec::OFFSET_S ; // Segmented, always zero for unsegmented data
+    header |= uint64_t(0x0)<<AMCHeaderSpec::OFFSET_M ; // More data
+    header |= uint64_t(0x0)<<AMCHeaderSpec::OFFSET_L ; // Indicates length error
+
+    AMCHeaders.push_back( header );
+
+  }
+
+  void addUHTR( uhtrData uhtr , uint64_t crate , uint64_t slot ){
+    // push uhtr data into FED container
+    uhtrs.push_back(uhtr);
+    // create the corresponding AMC header
+    addAMCHeader( crate , slot , uhtr.size()/4 );
+  };
+
+  // does not include HEADER and TRAILER
+  FEDRawData* formatFEDdata(){
+
+    //std::vector<unsigned char> output;
+    if( uhtrs.size() != AMCHeaders.size() ){
+      return NULL ;
+    }
+
+    // put common data format header in fed container
+    fedData.push_back((cdfHeader>>0 )&0xFF);
+    fedData.push_back((cdfHeader>>8 )&0xFF);
+    fedData.push_back((cdfHeader>>16)&0xFF);
+    fedData.push_back((cdfHeader>>24)&0xFF);
+    fedData.push_back((cdfHeader>>32)&0xFF);
+    fedData.push_back((cdfHeader>>40)&0xFF);
+    fedData.push_back((cdfHeader>>48)&0xFF);
+    fedData.push_back((cdfHeader>>56)&0xFF);
```

Comment:
write a loop to reduce the amount of copy-paste.
A (local namespace, can be inline) function that does this for a uint64_t argument would be even better, it would reduce the need to write 3 loops for cdfHeader, AMC13Headerm and AMCHeaders below